api/coreservice.go

```diff
@@ -1379,25 +1384,45 @@ func (core *coreService) LogsInRange(filter *logfilter.LogFilter, start, end, pa
 		return nil, err
 	}
 
-	// TODO: improve using goroutine
 	if paginationSize == 0 {
 		paginationSize = 1000
 	}
 	if paginationSize > 5000 {
 		paginationSize = 5000
 	}
+
+	c := make(chan _logsInBlock)
 	logs := []*iotextypes.Log{}
+	if len(blockNumbers) == 0 {
+		return logs, nil
+	}
 	for _, i := range blockNumbers {
```

Comment:
this is not practical, because `len(blockNumbers)` could be as large as 5000.

Reply:
But this is how we at the original

Reply:
How about set a max restrict `blockNumbers`, like 500 or something.

Reply:
> How about set a max restrict `blockNumbers`, like 500 or something.

you could write an uint test to test whether #goroutine will run out of cpu resources

Reply:
@CoderZhi 
How about this producer-consumer model, this should prevent go routines running out of cpu resources.
https://github.com/iotexproject/iotex-core/blob/a25b77f1bc0747efd713e62a1da8cca9746ad11f/api/coreservice.go#L1401-L1439